pkgs/by-name/bo/bob-nvim/package.nix

```
@@ -0,0 +1,41 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  installShellFiles,
+  stdenv,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "bob";
+  version = "4.1.6";
+
+  src = fetchFromGitHub {
+    owner = "MordechaiHadad";
+    repo = "bob";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-XI/oNGKLXQ/fpB6MojhTsEgmmPH1pHECD5oZgc1r4rQ=";
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  cargoHash = "sha256-YSZcYTGnMnN/srh8Z15toq+GIyRKfFd+pGkFQl5gCuo=";
+
+  doCheck = false;
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd bob \
+      --bash <($out/bin/bob complete bash) \
+      --fish <($out/bin/bob complete fish) \
+      --zsh <($out/bin/bob complete zsh) \
+      --nushell <($out/bin/bob complete nushell)
+  '';
```

```suggestion
  postInstall = let
    bob = "${stdenv.hostPlatform.emulator buildPackages} $out/bin/bob${stdenv.hostPlatform.extensions.executable}";
  in
    lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
      installShellCompletion --cmd bob \
        --bash <(${bob} complete bash) \
        --fish <(${bob} complete fish) \
        --zsh <(${bob} complete zsh) \
        --nushell <(${bob} complete nushell)
    '';
```
I've seen this kind of pattern used before (e.g. in [ripgrep](https://github.com/NixOS/nixpkgs/blob/8366a569873219932fd9ee93065859b16f85e554/pkgs/by-name/ri/ripgrep/package.nix#L15-L16)), might be worth adding it here.